feat: bulk CIDs import

MattWong-ca opened the pull request:
Introduces bulk import of CIDs using a text file, as requested in #2173 

Users can upload a text file with a list of CIDs (and an optional name right after), as shown below:
```
bafkreibm6jg3ux5qumhcn2b3flc3tyu6dmlb4xa7u5bf44yegnrjhc4yeq
QmawceGscqN4o8Y8Fv26UUmB454kn2bnkXV5tEQYc4jBd6 barrel.png
QmbvrHYWXAU1BuxMPNRtfeF4DS2oPmo5hat7ocqAkNPr74 pi equals.png
```
- Click `Bulk import` from Import dropdown and a modal to select a text file will pop up
- Frontend validation is done on each line of the text file using `isIPFS.cid()`, if any line fails then user won't be able to import it. Otherwise the name of the selected file will be shown
- Added new function `doFilesBulkCidImport` in actions.js responsible for going through text file and adding each CID
- Added stories.js file 
- Didn't include a CLI tutor mode command since it seemed more complex than the other ones already present, but happy to revisit if needed!

Testing
- Manually tested to ensure it works in both the root folder and subfolders
- Tried using a text file with improper formatting, which didn't let me import
- **NOTE:** the most number of CIDs I tested with in a file was 91, at which there was a slight lag before all the files were imported

Demo

https://github.com/user-attachments/assets/0eaee9ca-6791-40a9-8eda-80768c5e52fb

MattWong-ca commented:
Hey @SgtPooki, I've updated the PR description and added a video, this is good to go for review!

SgtPooki commented on src/files/modals/bulk-import-modal/BulkImportModal.js:
```diff
@@ -0,0 +1,125 @@
+  }
+}
+
+export default withTranslation('files')(BulkImportModal)
```
use `useTranslation('files')` once this is converted to a functional component instead of `withTranslation`

SgtPooki commented on src/files/modals/bulk-import-modal/BulkImportModal.js:
```diff
@@ -0,0 +1,125 @@
+import Icon from '../../../icons/StrokeDocument.js'
+import { normalizeFiles } from '../../../lib/files.js'
+
+class BulkImportModal extends React.Component {
```
we should convert this to a functional component.

SgtPooki commented on src/files/modals/bulk-import-modal/BulkImportModal.js:
```diff
@@ -0,0 +1,125 @@
+import { normalizeFiles } from '../../../lib/files.js'
+
+class BulkImportModal extends React.Component {
+  static propTypes = {
+    onCancel: PropTypes.func.isRequired,
+    onBulkCidImport: PropTypes.func.isRequired,
+    className: PropTypes.string
+  }
```
we should remove prop-types. Eventually, we're moving this whole repo to typescript and will get type safety during transpilation.

SgtPooki commented on src/files/modals/bulk-import-modal/BulkImportModal.js:
```diff
@@ -0,0 +1,125 @@
+import React from 'react'
```
We should use typescript for new files

SgtPooki reviewed (changes requested):
this is great, thanks so much! Only change I can see that is needed is the removal of prop-types, and having the new `BulkImportModal` be a functional typescript component.

Also, we should use kebab-case-filenames instead of CamelCase. This repo hasn't been fully updated, but that's where we're going

MattWong-ca commented:
> this is great, thanks so much! Only change I can see that is needed is the removal of prop-types, and having the new `BulkImportModal` be a functional typescript component.
> 
> Also, we should use kebab-case-filenames instead of CamelCase. This repo hasn't been fully updated, but that's where we're going

Thanks @SgtPooki, just finished addressing all the comments!

MattWong-ca commented:
Hey @SgtPooki, wanted to bump this!

SgtPooki reviewed (commented):
code looks okay, let me check locally

SgtPooki commented:
this worked great... let me merge main and give you an approval
